From d210de059112101903a61ffa3213ea701980978d Mon Sep 17 00:00:00 2001
From: Jeff Kudrick <jeff.kudrick@freescale.com>
Date: Thu, 14 Jun 2012 13:19:33 -0500
Subject: [PATCH 292/635] ENGR00213381: MJPEG decode not working in a majority
 of cases

One of the several problems surrounding this issues was failure to
support buffer allocation in gralloc for HAL_PIXEL_FORMAT_YCbCr_420_P
and HAL_PIXEL_FORMAT_YCbCr_422_P.  The support for both of the pixel
formats has been added to the alloc() routine.

Signed-off-by: Jeff Kudrick <jeff.kudrick@freescale.com>
---
 mx5x/libgralloc/gralloc.cpp | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/mx5x/libgralloc/gralloc.cpp b/mx5x/libgralloc/gralloc.cpp
index dccbe28..93b1ca8 100755
--- a/mx5x/libgralloc/gralloc.cpp
+++ b/mx5x/libgralloc/gralloc.cpp
@@ -369,8 +369,9 @@ static int gralloc_alloc(alloc_device_t* dev,
 
     size_t size, alignedw, alignedh;
     if (format == HAL_PIXEL_FORMAT_YCbCr_420_SP || format == HAL_PIXEL_FORMAT_YCbCr_422_I ||
-            format == HAL_PIXEL_FORMAT_YCbCr_422_SP || format == HAL_PIXEL_FORMAT_YCbCr_420_I ||
-            format == HAL_PIXEL_FORMAT_YV12)
+        format == HAL_PIXEL_FORMAT_YCbCr_422_SP || format == HAL_PIXEL_FORMAT_YCbCr_420_I ||
+        format == HAL_PIXEL_FORMAT_YV12         || format == HAL_PIXEL_FORMAT_YCbCr_420_P ||
+        format == HAL_PIXEL_FORMAT_YCbCr_422_P)
     {
         int luma_size;
         int chroma_size;
@@ -385,10 +386,12 @@ static int gralloc_alloc(alloc_device_t* dev,
         switch (format) {
             case HAL_PIXEL_FORMAT_YCbCr_422_SP:
             case HAL_PIXEL_FORMAT_YCbCr_422_I:
+            case HAL_PIXEL_FORMAT_YCbCr_422_P:
                 chroma_size = ALIGN_PIXEL_4096( (alignedw * alignedh) / 2) * 2;
                 break;
             case HAL_PIXEL_FORMAT_YCbCr_420_SP:
             case HAL_PIXEL_FORMAT_YCbCr_420_I:
+            case HAL_PIXEL_FORMAT_YCbCr_420_P:
             case HAL_PIXEL_FORMAT_YV12:
                 chroma_size = ALIGN_PIXEL_4096(alignedw/2 * alignedh/2) * 2;
                 break;
-- 
1.8.0

